evalml/data_checks/invalid_targets_data_check.py

```diff
@@ -87,17 +92,41 @@ def validate(self, X, y):
                 details = {"target_values": unique_values}
             else:
                 details = {"target_values": unique_values[:min(self.n_unique, len(unique_values))]}
-            messages["errors"].append(DataCheckError(message="Target does not have two unique values which is not supported for binary classification",
+            messages["errors"].append(DataCheckError(message="Binary class targets require exactly two unique values.",
                                                      data_check_name=self.name,
                                                      message_code=DataCheckMessageCode.TARGET_BINARY_NOT_TWO_UNIQUE_VALUES,
                                                      details=details).to_dict())
-        if self.problem_type == ProblemTypes.MULTICLASS and value_counts.min() <= 1:
-            least_populated = value_counts[value_counts <= 1]
-            details = {"least_populated_class_labels": least_populated.index.tolist()}
-            messages["errors"].append(DataCheckError(message="Target does not have at least two instances per class which is required for multiclass classification",
+
+        if self.problem_type == ProblemTypes.REGRESSION and not pd.api.types.is_numeric_dtype(y.to_series()):
+            messages["errors"].append(DataCheckError(message="Target data type should be floating point for regression type problems.",
```

Comment:
Wouldn't integers be allowed here as well? I believe `is_numeric_dtype` will return True regardless of if the dtype is `Int64` or `float64`. Maybe the message should be `Target data type should be numerical for regression type problems.`

Reply:
Agree.  I definitely meant to put that initially and I got distracted.  Thanks for the catch.